tests/test_aampi.py

```diff
@@ -1146,3 +1146,64 @@ def test_aampi_self_join_egress_KNN():
                 npt.assert_almost_equal(ref_I, comp_I)
                 npt.assert_almost_equal(ref_left_P, comp_left_P)
                 npt.assert_almost_equal(ref_left_I, comp_left_I)
+
+
+def test_aampi_self_join_egress_passing_mp():
+    m = 3
+
+    for p in [1.0, 2.0, 3.0]:
+        seed = np.random.randint(100000)
+        np.random.seed(seed)
+
+        n = 30
+        T = np.random.rand(n)
+        mp = naive.aamp(T, m, p=p)
+
+        ref_mp = naive.aampi_egress(T, m, p=p, mp=mp)
+        ref_P = ref_mp.P_.copy()
+        ref_I = ref_mp.I_
+        ref_left_P = ref_mp.left_P_.copy()
+        ref_left_I = ref_mp.left_I_
+
+        stream = aampi(T, m, egress=True, p=p, mp=mp)
+
```

Comment:
No sure if this is a good way to test the new param `mp`. I pass `mp` to both naive and performant `floss`. Alternatively, we can just pass it to performant `floss`, and then compare it with naive version when `mp` is computed internally.

Reply:
Passing `mp` to both seems reasonable